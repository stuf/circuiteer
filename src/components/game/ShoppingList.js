import * as P from 'prop-types';

export function ShoppingList(props) {
  const { entities } = props;

  return <div>shopping list</div>;
}

ShoppingList.propTypes = {
  entities: P.arrayOf(P.object),
};

ShoppingList.defaultProps = {
  entities: [],
};
