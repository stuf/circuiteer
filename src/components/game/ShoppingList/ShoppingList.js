import { useTranslation } from 'react-i18next';
import * as R from 'ramda';
import * as P from 'prop-types';

import { shoppingList } from 'core/data/meta/shopping-list';
import { thru } from 'common/util';
import { Group } from 'components';

export function ShoppingList(props) {
  const { t } = useTranslation();

  const { entities } = props;
  const modules = entities.map(it => it.module);
  const list = shoppingList(modules);
  const items = Object.entries(list);

  return (
    <Group title={t('ui:section.shoppingList.title')}>
      <dl className="divide-y-2 divide-dashed text-sm">
        {thru(
          items,
          R.sortBy(R.prop(0)),
          R.map(([material, count]) => (
            <div className="grid grid-cols-4 py-1" key={material}>
              <dt
                id={`shoppinglist-${material}`}
                aria-label={t(`game:material.${material}`)}
                className="font-normal col-span-3"
              >
                {t(`game:material.${material}`)}
              </dt>
              <dd
                aria-labelledby={`shoppinglist-${material}`}
                className="font-mono text-right"
              >
                {count}
              </dd>
            </div>
          )),
        )}
      </dl>
    </Group>
  );
}

ShoppingList.propTypes = {
  entities: P.arrayOf(P.object),
};

ShoppingList.defaultProps = {
  entities: [],
};
