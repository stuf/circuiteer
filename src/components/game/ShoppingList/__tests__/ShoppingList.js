import { render, logRoles } from '@testing-library/react';
import { ShoppingList } from '../ShoppingList';

test('ShoppingList', async () => {
  const props = {
    entities: [{ module: 'smallWindTurbine' }],
  };

  const { container } = render(<ShoppingList {...props} />);

  expect(container).toMatchSnapshot();
});
