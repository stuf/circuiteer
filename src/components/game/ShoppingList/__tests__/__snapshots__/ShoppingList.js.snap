// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingList 1`] = `
<div>
  <section>
    <header
      aria-label="ui:section.shoppingList.title"
      class="font-bold mb-2"
    >
      ui:section.shoppingList.title
    </header>
    <div
      class="space-y-2"
    >
      <dl
        class="divide-y-2 divide-dashed text-sm"
      >
        <div
          class="grid grid-cols-4 py-1"
        >
          <dt
            aria-label="game:material.ceramic"
            class="font-normal col-span-3"
            id="shoppinglist-ceramic"
          >
            game:material.ceramic
          </dt>
          <dd
            aria-labelledby="shoppinglist-ceramic"
            class="font-mono text-right"
          >
            1
          </dd>
        </div>
      </dl>
    </div>
  </section>
</div>
`;
